swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SettingsView.swift

```diff
@@ -377,7 +376,7 @@ public struct SettingsView: View {
                   favorites.reset()
                 }
               )
-              .disabled(favorites.ids.isEmpty && model.settings == Settings.defaultValue)
+              .disabled(favorites.isEmpty() && model.settings == Settings.defaultValue)
```

Comment:
IMO, this logic here should be in the `SettingsViewModel`. All you want to "ask" the viewmodel is, `isResetToDefaultsDisabled`. That leaves basically no room for bugs in terms of behaviour if you were to have a unit test that asserts that boolean under multiple conditions.

I think Robert Martin explains that concept quite well in this talk: https://www.youtube.com/watch?v=o_TH-Y78tt4

Reply:
Yeah, I can do the refactoring in this PR.

Reply:
Uncle bob is getting quite old! ;-)

Reply:
> Uncle bob is getting quite old! ;-)

He is and a bit controversial at times but in this particular video, he does a great job IMO.